### PR TITLE
[WINDOW_MANAGER] Adds display server protocol to entry

### PR DESCRIPTION
**Pull request by HorlogeSkynet**
See #142.

## How has this been tested ?

Unit tests and locally (Debian 11).

## Types of changes :

- Bug fix (non-breaking change which fixes an issue)
- Typo / style fix (non-breaking change which improves readability)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist :

- \[IF NEEDED\] I have updated the _README.md_ file accordingly ;
- [X] \[IF NEEDED\] I have updated the test cases (which pass) accordingly ;
- \[IF BREAKING\] This pull request targets next Archey version branch ;
- [X] My changes looks good ;
- [X] I agree that my code may be modified in the future ;
- [X] My code follows the code style of this project ([PEP8](https://www.python.org/dev/peps/pep-0008/)).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,11 @@ and this project (partially) adheres to [Semantic Versioning](https://semver.org
 ## [Unreleased]
 ### Added
 - Official EndeavourOS distribution support
+- Display server protocol to `WindowManager`
 
 ### Changed
 - Allow `df` output to contain Unicode characters
+- `WindowManager` API value format (now an object)
```
Comment (ingrinder) [inline CHANGELOG.md:15]:
```suggestion
- `WindowManager` API value format: now an object with `name` and `display_server_protocol` attributes
```

Worth being more specific?

**Review by ingrinder — APPROVED** [review]
Clean :tada:! LGTM, possibly a bit more detail for the changelog if you agree?

File: CHANGELOG.md (hunk 1 above)
Comment (HorlogeSkynet) [reply CHANGELOG.md:15]: Yes, it would be better. I wonder whether DSP should rather be added to `DestktopEnvironment` entry though 🤔

**Comment by HorlogeSkynet** [comment]
So I've checked how UI layering works and it appears showing DSP info in DE entry would be more consistent. Tell me what you think about it @Tobilike @ingrinder :pray:

**Comment by ingrinder** [comment]
Hmmm, how come? As far as I know you can run independent window managers without a "DE" as such in X and Wayland (e.g. `i3`, `weston`), so does it not make more sense there?

**Comment by Tobilike** [comment]
I tested it on Hyprland and xmonad. It is displayed in the correct place. As long as it is displayed in windows manager. Everything is fine.
Technically, you can run x11 without everything, but it doesn't really make sense. You can't do anything.

Very nice work by the way. It's much better integrated.

**Comment by HorlogeSkynet** [comment]
Thanks for your feedbacks (I am not an expert in UI environments at all). I've reverted to first proposal (including your CHANGELOG suggestion Michael).

Bye :wave:

**Pull request merged**